### PR TITLE
Added documentation of type system.

### PR DESCRIPTION
**Pull request by jesper-friis**
# Description:
Added documentation of type system.

## Type of change:

- [ ] Bug fix.
- [ ] New feature.
- [x] Documentation update.

## Checklist for the reviewer:

This checklist should be used as a help for the reviewer.

- [ ] Is the change limited to one issue?
- [ ] Does this PR close the issue?
- [x] Is the code easy to read and understand?
- [ ] Do all new feature have an accompanying new test?
- [x] Has the documentation been updated as necessary?

### CODE REVIEW

File: doc/user_guide/type-system.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+Type system
+===========
+
+Simple unified access to all data types
+---------------------------------------
+The datamodel API for accessing properties of an instance in SOFT, has
```
Comment (alfredoisg) [inline doc/user_guide/type-system.md:6]:
```suggestion
The datamodel API for accessing properties of an instance in SOFT has
```

File: doc/user_guide/type-system.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+separate getters and setters for each type and number of dimensions.
+DLite generalize and simplifies this by describing types and
```
Comment (alfredoisg) [inline doc/user_guide/type-system.md:8]:
```suggestion
DLite generalizes and simplifies this by describing types and
```

File: doc/user_guide/type-system.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+dimensionality of properties with 4 parameters:
+
+  - `type`: an enum defining the type of the data item (or items if
+    it has dimensions), e.g. whether it is an integer, float or string...
```
Comment (alfredoisg) [inline doc/user_guide/type-system.md:12]: "e.g. whether it is an integer, float or string.."
> I do not think, this explanation is needed, given the table below

File: doc/user_guide/type-system.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    The table below summarises the implemented dtype's.
+  - `size`: the size of a data item in bytes.
+  - `ndims`: number of dimensions.  Scalars has ``ndims=0``.
+  - `dims`: array of length `ndims` with the length of each dimension.
+
+By taking these parameters as arguments, the functions
+DLite_datamodel_get_property() and DLite_datamodel_set_property() can handle
```
Comment (alfredoisg) [inline doc/user_guide/type-system.md:19]: What about [dlite_datamodel_get_property()](https://github.com/SINTEF/dlite/blob/master/src/dlite-datamodel.c#L118) and [dlite_datamodel_set_property()](https://github.com/SINTEF/dlite/blob/master/src/dlite-datamodel.c#L134)

(Or the corresponding link to the definitions)

File: doc/user_guide/type-system.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+all supported property types.  No storage strategy is needed.
+
+The table below summarises the different dtypes defined in DLite.  For
+more details, see dlite-type.h.  Also note that this supports arbitrary
```
Comment (alfredoisg) [inline doc/user_guide/type-system.md:23]: [dlite-type.h](https://github.com/SINTEF/dlite/blob/master/src/dlite-type.h) ?

File: doc/user_guide/type-system.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+uint      | dliteUInt      | 1, 2, 4, {8}           | unsigned integer                 | (uint), uint8, uint16, uint32, {uint64}
+float     | dliteFloat     | 4, 8, {10, 12, 16}     | floating point                   | (float), (double), float32, float64, {float80, float96, float128}
+fixstring | dliteFixString | any                    | fix-sized NUL-terminated string  | string20, string4000, ...
+ref       | dliteRef       | sizeof(DLiteInstance*) | reference to another instance    | ref, http://onto-ns.com/meta/0.1/MyEntity
+string    | dliteStringPtr | sizeof(char *)         | pointer to NUL-terminated string | string
+relation  | dliteRelation  | sizeof(DLiteRelation)  | subject-predicate-object triplet | relation
+dimension | dliteDimension | sizeof(DLiteDimension) | only intended for metadata       | dimension
+property  | dliteProperty  | sizeof(DLiteProperty)  | only intended for metadata       | property
+
+The examples shown in curly parenthesis may not be supported on all
```
Comment (alfredoisg) [inline doc/user_guide/type-system.md:42]:
```suggestion
The examples shown in curly brackets may not be supported on all
```

File: doc/user_guide/type-system.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+platforms.  The size int, uint, float and double are
+platform-dependent.  For portable applications you should to provide
```
Comment (alfredoisg) [inline doc/user_guide/type-system.md:44]:
```suggestion
platform-dependent.  For portable applications you should provide
```

**Review by alfredoisg — APPROVED** [review]
Very nice! Here are some suggestions.

File: doc/user_guide/type-system.md (hunk 4 above)
Comment (jesper-friis) [reply doc/user_guide/type-system.md:19]: definitely, these were very old names

**Pull request merged**